src/lava/proc/lif/process.py

```diff
@@ -28,20 +50,39 @@ class LIF(AbstractProcess):
     """
     def __init__(self, **kwargs):
         super().__init__(**kwargs)
-        shape = kwargs.get("shape", (1,))
-        du = kwargs.pop("du", 0)
-        dv = kwargs.pop("dv", 0)
-        bias = kwargs.pop("bias", 0)
-        bias_exp = kwargs.pop("bias_exp", 0)
         vth = kwargs.pop("vth", 10)
 
-        self.shape = shape
-        self.a_in = InPort(shape=shape)
-        self.s_out = OutPort(shape=shape)
-        self.u = Var(shape=shape, init=0)
-        self.v = Var(shape=shape, init=0)
-        self.du = Var(shape=(1,), init=du)
-        self.dv = Var(shape=(1,), init=dv)
-        self.bias = Var(shape=shape, init=bias)
-        self.bias_exp = Var(shape=shape, init=bias_exp)
         self.vth = Var(shape=(1,), init=vth)
+
```

Comment:
Rationale for having one threshold is same as that for du and dv. These variables are set using compartment config in Loihi 1. There are only 32 of these configurations per neurocore. So 32 populations can have different thresholds per neurocore. This means, we can create different instance of LIF corresponding to those populations. This will be different for Loihi 2, but we can refactor a little bit in the future.

Reply:
Given the Loihi1 constraints, I think it's arguably clearer to make different LIF Population for different constants anyways. 
The best counter to this rationale would be a scenario in which your connectivity pattern is population dependent and population groupings for connectivity are not 1:1 with the subsets of neurons that have different du/dv/v_th constants.

However, we probably want to re-evaluate this requirement for Loihi2, where you might want to implement something like an adaptive threshold neuron program in microcode and every neuron will require its own threshold. 

I would say in the next refactor, we should allow heterogeneous du, dv, v_th within in a LIF process, in anticipation of the Loihi2 compiler.

Reply:
So, if the user wants to use the 32 possible configurations for threshold but let the other parameters the same, the user will have to create 32 LIF populations rather than a single population with 32 threshold values. I think that may not be the best user experience. But we can discuss it further on a following refactoring ;)